client/src/modules/dashboard/Venues/utils.ts

```diff
@@ -0,0 +1,34 @@
+import { useRouter } from 'next/router';
+import { useCreateVenueMutation } from '../../../generated/graphql';
+import { VenueFormData } from './components/VenueForm';
+import { VENUES } from './graphql/queries';
+
+export const useSubmitVenue = (setLoading: (x: boolean) => void) => {
+  const [createVenue] = useCreateVenueMutation({
+    refetchQueries: [{ query: VENUES }],
+  });
+  const router = useRouter();
+
+  return async (data: VenueFormData) => {
+    setLoading(true);
+    const { chapter_id, ...createData } = data;
+    try {
+      const latitude = parseFloat(String(data.latitude));
+      const longitude = parseFloat(String(data.longitude));
+
+      const venue = await createVenue({
+        variables: {
+          chapterId: chapter_id,
+          data: { ...createData, latitude, longitude },
+        },
+      });
+      if (venue.data) {
+        router.replace(`/dashboard/venues/${venue.data.createVenue.id}`);
+      }
+    } catch (err) {
+      console.error(err);
+    } finally {
+      setLoading(false);
+    }
```

Comment:
A bit side note, from the observing form behavior when it's submitted, for the other comment. When submission is successful there's a moment before redirect, when loading is set back to `false`, this can be seen in the form and submit button (not using loading text anymore).

That made me wonder if setting loading always back to `false` is desired, after all if there's redirection it shouldn't matter. Only if we are staying on the page enabling form again is needed. I've looked on some other pages and few `await`s for the `router.replace` what makes the issue not relevant.

Reply:
I see what you mean and the current behaviour doesn't seem quite right.

It's a touch better if it back to `false` if there's an error, but leave it otherwise (as you say, it's pointless if we're redirecting).  At least they can change something and try again. It's still bad UX, but slightly less bad than if the form becomes obviously stuck.